### PR TITLE
Convert LessThanInstruction classes closer to .NET Core implementation

### PR DESCRIPTION
**Pull request by gpetrou**

### CODE REVIEW

File: Src/Microsoft.Dynamic/Interpreter/Instructions/LessThanInstruction.cs [hunk 1]
```diff
@@ -8,118 +8,238 @@
 using Microsoft.Scripting.Utils;
 
 namespace Microsoft.Scripting.Interpreter {
-    public abstract class LessThanInstruction : Instruction {
+    internal abstract class LessThanInstruction : Instruction {
+        public static readonly object BoxedFalse = false;
+        public static readonly object BoxedTrue = true;
```
Comment (gpetrou) [inline Src/Microsoft.Dynamic/Interpreter/Instructions/LessThanInstruction.cs]: Where should we place the above two lines? In a new class somewhere?

Comment (slozier) [reply Src/Microsoft.Dynamic/Interpreter/Instructions/LessThanInstruction.cs]: There are ScriptingRuntimeHelpers.True/False which we should use instead.

**Comment by slide** [comment]
Do they have additional tests for these things as well?

**Comment by slide** [comment]
I found this: https://github.com/dotnet/corefx/blob/master/src/System.Linq.Expressions/tests/InterpreterTests.cs

**Comment by gpetrou** [comment]
Looking through .NET Core repository, I don't see dedicated tests to Instruction classes, such as the one in this PR.
Should we add the Interpreter test in a separate PR?

**Comment by slozier** [comment]
I don't think we can use the InterpreterTests from .NET Core as is. Better to leave it for another PR. We should look into creating some basic tests for the instructions though.

File: Src/Microsoft.Dynamic/Interpreter/Instructions/LessThanInstruction.cs (hunk 1 above)
Comment (slide) [reply Src/Microsoft.Dynamic/Interpreter/Instructions/LessThanInstruction.cs]: @gpetrou Are you still wanting this merged? Can you update to use the ScriptingRuntimeHelpers?

Comment (gpetrou) [reply Src/Microsoft.Dynamic/Interpreter/Instructions/LessThanInstruction.cs]: @slide I did as you asked. Do you want to merge this?

**Comment by slozier** [comment]
@gpetrou Thanks for the PR, sorry it took so long to get merged in.

**Pull request merged**